### PR TITLE
Feature/typescript

### PR DESCRIPTION
**Pull request by idodeclare**
Hello,

Please consider for integration this patch to add TypeScript support based on TypeScript 3.6 and ECMAScript 10th Edition, June 2019.

* Add ECMAScript.lexh and ECMAScriptProductions.lexh.
* `.TS` extension is re-mapped to `TypeScriptAnalyzerFactory`.
* `AnalyzerGuru` version is bumped so that existing, indexed `.TS` files are re-analyzed by the new `TypeScriptAnalyzer`.
* Refactor `JavaScriptAnalyzer` also to use the 10th Edition ECMAScript*.lexh files.
* `JavaScriptAnalyzer`version is bumped so that existing, indexed `.JS` files are re-analyzed.

Thank you.

### CODE REVIEW

File: opengrok-indexer/src/test/resources/analysis/typescript/sample.ts [hunk 1]
```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright 2019 GitHub
```
Comment (tarzanek) [inline opengrok-indexer/src/test/resources/analysis/typescript/sample.ts]: hmm, below is MIT ? if yes I am OK

**Comment by tarzanek** [comment]
nice, thank you very much for this, I'd think it can be merged, assuming all licensing is in order
@vladak what do you think?

File: opengrok-indexer/src/test/resources/analysis/typescript/sample.ts (hunk 1 above)
Comment (idodeclare) [reply opengrok-indexer/src/test/resources/analysis/typescript/sample.ts]: Yes. I scrounge for substantial samples that are CDDL-compatible

**Comment by tarzanek** [comment]
thank you @idodeclare !

**Comment by idodeclare** [comment]
@tarzanek , my pleasure!

**Pull request merged**